database/migrations/2022_12_02_170921_add_warranty_expires_to_assets.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+class AddWarrantyExpiresToAssets extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('assets', function (Blueprint $table) {
+            $table->date('warranty_expires')->virtualAs('DATE_ADD(`purchase_date`, INTERVAL `warranty_months` MONTH)')->nullable()->default(null);
```

Comment:
I'd much rather we just store this as a date column that we populate based on the other information we have when saving, or perhaps even allow the user to override manually. Virtual columns are not supported by some of the other SQL database types that we have to unofficially support. If we want to sort on a date, we should just store it as a regular date and do the math for what that date should be in the code.

Reply:
Thanks! I will work on doing it that way instead.